src/util/namespace.h

```diff
@@ -84,8 +84,20 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`. If a symbol is
+  /// found it will be returned in \p symbol.
+  /// \return False iff the requested symbol is found in at least one of the two
+  /// tables
```

Comment:
Many thanks. I took your suggestion, moved the comments to `namespace_baset`, which now looks like this:

```C++
  /// Returns the maximum integer x such that there is a symbol (in some of the
  /// two symbol tables) whose name has \p prefix as prefix and x as suffix.
  /// The intended use case is finding the next available symbol name for a
  /// sequence of auto-generated symbols.
  /// This method expects that all symbols whose name starts with \p prefix to
  /// have a name of the form "prefixNNN" where NNN is an integer. Note that
  /// NNN is parsed using unsafe_string2unsigned(), which returns 0 if NNN
  /// isn't an integer.
  virtual unsigned get_max(const std::string &prefix) const=0;

  /// Searches for a symbol named \p name. Iff found, set \p symbol to point to
  /// the symbol and return false; else \p symbol is unmodified and `true` is
  /// returned. With multiple symbol tables, `symbol_table1` is searched first
  /// and then symbol_table2.
  /// \return False iff the requested symbol is found in at least one of the
  /// tables.
  virtual bool lookup(const irep_idt &name, const symbolt *&symbol) const=0;
```